### ISSUE TITLE
Spotlight hour calendar not using correct local time

### ISSUE THREAD

**Issue by ShahrumAmiri** [othyn/go-calendar#6]
### Description

Hello,

I’m currently subscribed to the .ics feed (via Google Calendar) for the spotlight hour calendar and the events are coming through at 2PM EDT instead of 6PM EDT, the expected time for my timezone. Is there a configuration I’m missing?

### Version

Latest

### Steps to Reproduce

Add calendar feed to Google Calendar
Events loaded
Look at calendar and see events at wrong local time (expected 6pm EDT, seen at 2pm EDT)

### Device

All

### Operating System

Latest

**Comment by PokeIMGs** [comment]
Having similar issues.

What I found is just importing the ICS file directly into another calender works fine, but subscribing sets the times out of whack.

I am GMT + 1 (BST)

Spotlight Hours are an hour ahead
Raid Hours are 7 hours ahead
Go fest global is an hour ahead
Community day is 5 hours ahead
City Sefari Seoul and Barcelona are an hour ahead, but Mexico is the right time (funilly enough this is after daylight savings ends)

**Comment by ShahrumAmiri** [comment]
I like subscribing so it will update for me automatically. I imagine an
import means you have to update it yourself by reimporting periodically?

**Comment by PokeIMGs** [comment]
Yup, I would also prefer to use the subscription for those reasons but am
at a loss as to why the times are so different

**Comment by kiararey** [comment]
I'm having a similar issue. I'm on central time, and the following are at different times for me:

Events calendar synced fine I think, though Noxious Swamp should end on Tuesday and not Monday
Spotlight hours: show up 5 hours early
Raid hours: show up 1 hour early
GO fest: show up 5 hours early
I also subscribed to the community day calendar but there's currently no event for me to check if it works correctly

**Comment by othyn** [comment]
This is the same issue as #4, it appears to be a long standing issue specifically with Google Calendar's processing of the feed.

I have no idea why it does it and all the fixes I've put in appear to have made no difference. I purposefully do not embed a TZ within the generated files so that the consuming client will use its own and correctly translate it into the local TZ.

#5 aimed to resolve this issue, but I haven't had time yet to look into it properly as its a large amount of infra work, basically to generate versions for all timezones on the fly, which appears to be the only way to have it work with Google Calendar. Which is what I wanted to avoid at all costs as it adds a ton of complexity.

**Comment by othyn** [comment]
No specific recommendations, I just use the default Apple ones too.

Google Calendar appears to be the only calendar affected, based on user reports of this issue.

**Comment by gaylie** [comment]
Hello. Poking in to say that the issue still seems to exist. Also using Google Calendar, the times for Spotlight hours show up at 8PM-9PM for me

**Comment by othyn** [comment]
Not really sure what else I can do on my side, I'd love someone's input with more knowledge or experience with Google Calendar event generation from a dev perspective.

Someone submitted an amazing PR for dynamic generation of calendars for each timezone, however its a bit of a heavy handed solution and it _feels_ like there should be an easier solve that I'm just not seeing.

It appears to _only_ be a Google Calendar issue with a really weird and non-industry standard interpretation (from what I can see, would like to be corrected on that!) of the generated events. What is extra weird is that it only appears to be a problem for _subscriptions_, if you manually import the calendar as a one-off into Google Calendar then its fine and imports it with the local timezone interpreted as intended. All events are generated in a way in which forces there to be no timezone and its left to the local client to interpret the timezone.

When subscribing to it, Google just forces the timezone to be `(GMT+00:00) Coordinated Universal Time` instead of interpreting it into the users timezone like all other clients do, and you are not given the option in the calendar settings to change it.

**Comment by othyn** [comment]
On a hunch, I've tried something and it appears to have worked in my testing.

Can people subscribe to the temporary calendar I've created with a potential fix in here and report back?

https://raw.githubusercontent.com/othyn/go-calendar/gcal-sub-fix/dist/gocal.ics

Please let me know if that subscribes correctly. If it does, I'll have to publish a set of 'If you're using gcal, use this link instead' files specifically for Google Calendar.

**Comment by gaylie** [comment]
> On a hunch, I've tried something and it appears to have worked in my testing. I've subscribed to the production one and this new one side by side, and the new one imports into the correct timezone (client) whereas the old one appears an hour after it should.
> 
> Can people subscribe to the temporary calendar I've created with a potential fix in here and report back?
> 
> https://raw.githubusercontent.com/othyn/go-calendar/gcal-sub-fix/dist/gocal.ics
> 
> Please let me know if that subscribes correctly?
> 
> If it does, I'll have to publish a set of 'If you're using gcal, use this link instead' files specifically for Google Calendar subscriptions.

Didn't expect such a quick response!

I subscribed to the link, though sadly it still seems slightly off (differently so than before, though).
For some examples: Spotlight Hours show up at 7PM-8PM for me, Pokestop Showcases from 11AM-9PM, Raid Hours from 7PM-8PM... Generally they seem an hour off for me.

That said, I am in Central European Summer Time (CEST) as of right now, and the times WOULD be correct for CET (Winter time). Perhaps Summer/Winter timezones are messing this up?

**Comment by PokeIMGs** [comment]
Yes! I can confirm this new calendar, when subscribed to is showing the
correct times!

Tested using Google cal URL subscription on British Timezone (BST/GMT+1)

**Comment by othyn** [comment]
> That said, I am in Central European Summer Time (CEST) as of right now, and the times WOULD be correct for CET (Winter time). Perhaps Summer/Winter timezones are messing this up?

That's like a timezone in a timezone problem haha although I'm currently in BST (UTC+1) and it is working, which is the equivalent summer offset.

I need some people in more timezones to test it. I want to make sure its a local timezone fix and not something else, such as just a +1 offset that Google Calendar is applying internally  (as CEST being UTC+2 is still leaving only an hour offset).

> Tested using Google cal URL subscription on British Timezone (BST/GMT+1)

Awesome, that aligns with my testing too in BST.

**Comment by kiararey** [comment]
> On a hunch, I've tried something and it appears to have worked in my testing. I've subscribed to the production one and this new one side by side, and the new one imports into the correct timezone (client) whereas the old one appears an hour after it should.
> 
> Can people subscribe to the temporary calendar I've created with a potential fix in here and report back?
> 
> https://raw.githubusercontent.com/othyn/go-calendar/gcal-sub-fix/dist/gocal.ics
> 
> Please let me know if that subscribes correctly?
> 
> If it does, I'll have to publish a set of 'If you're using gcal, use this link instead' files specifically for Google Calendar subscriptions.

this new calendar still shows the spotlight hours at 1pm on Google Calendar :(

**Comment by othyn** [comment]
Damn, okay. I don't think there is much more I can do then, it appears to be an issue specifically with Google Calendar subscription feeds and whatever they are doing to force/ignore timezones from what I can gather.

All other clients respect the standards and use the clients local timezone when one isn't specified, appears Google Calendar is the only one to not adhere to that rule, and also only on subscriptions - manual imports on Google Calendar work fine.

If anyone has any other ideas or fixes, I'm all ears. The complete re-write approach (PR) is awesome, but I'd rather not go that route if at all possible just due to whats involved. Especially as I don't play the game anymore either.

**Comment by KemptonM** [comment]
For what it's worth, Google Calendar used my default time zone (where I live) but it was incorrect for where I currently am, living abroad.

**Comment by Boby360** [comment]
To add to this, this is what is shown in Outlook.
The timezone is forced to UTC and I cannot change it when subscribing as it is read only.

My phone however shows it has my timezone, but still 6-9AM.
As this subscription is synced across devices I don't know if one thing is fighting the other, but they both show incorrect times.
It sounds like it was suggested to only be a google calendar issue, so figured id mention my observations.
These are synced through outlook on my hotmail email through an exchange server.

![image](https://github.com/othyn/go-calendar/assets/1607258/b8ca980b-5928-4027-9d45-ae4347265156)
![image](https://github.com/othyn/go-calendar/assets/1607258/9eb31735-c271-4de5-96d4-367b0b44a6b4)
![image](https://github.com/othyn/go-calendar/assets/1607258/785be40b-1968-4bf9-a253-48a9c17c2360)

**Comment by othyn** [comment]
This issue is the worst, and a prime example of [XKCD 927](https://xkcd.com/927/):

![image](https://github.com/othyn/go-calendar/assets/7256684/1f75040e-d899-4478-8ed2-9fe8dc5314b4)

Every. Single. Client. appears to handle the 'standard' timezone signature differently.

Some of them interpret the lack of a timezone correctly as 'use the local users timezone'. Some of them read it in as fixed UTC. Some of them read it in as an undefined timezone. Some of them read it in as whatever they feel like that day.

It varies per device, per client and per calendar permutation. A thorn in my side on what should have been a very simple 'please just use the users local timezone' implementation.

I may run this problem through ChatGPT - as that wasn't an option when this was created, as I've exhausted just about every single blog, docs and SO post imaginable on this problem to try and resolve it. Give the AI a run at it lmao

**Comment by Boby360** [comment]
I forked it and ran some versions through chatgpt but didn't have any improvement.

**Comment by othyn** [comment]
Yeah I had a rather lengthy back and forth with ChatGPT over this, feeding it different examples and trying to get it to generate its own approach and compare. It just confirms the information that I've already found at length on the topic - this _shouldn't_ be an issue and I still cannot understand why it is. The examples that it generates are _identical_ to the ones that this repo does.

The goal: Given a hypothetical event, let's say starts at 6:00 PM, users in London and New York will see the event at their respective local times. If a user in London downloads the calendar, they will see the event at 6:00 PM local time in London. If a user in New York downloads the calendar, they will see the event at 6:00 PM local time in New York.

The recommended fix, as is already implemented, is to _not_ define a timezone (e.g. `TZID=Europe/London`) within the `DTSTART` and `DTEND` components, at which point the calendar client should use floating time (without a specific timezone), adjusting to the user's local timezone.

That is how the current _live_ implementation of any calendar generated by this tool is generated, so I'm still at at a loss to why any of this is an issue.

Google Calendar, Outlook/O365, iOS/iPadOS/macOS (Apple Calendar), etc. all support 'floating time' as defined by the absence of a timezone (e.g. `TZID=Europe/London`) definition within the `DTSTART` and `DTEND` components, which is exactly what this repo does.

If someone can find a working example on the web for this sort of problem, then I can reverse engineer a solution. I did think I found one in the [Formula 1 calendar project](https://f1calendar.com/), but those events are fixed in time to each timezone, as an F1 race occurs at a fixed point in time. Pokemon GO events occur at the local time for each user, so its not a comparison that can be made.

**Comment by othyn** [comment]
I've just done another test. No matter what, Google Calendar will _always_ set the timezone of the calendar subscription to `(GMT+00:00) Coordinated Universal Time` with the field to change it being non-editable. Whereas, Apple Calendar correctly adheres to RFC 5545 and _doesn't_ impose a timezone on the subscribed calendar.

<img width="1389" alt="Screenshot 2024-02-04 at 23 48 35" src="https://github.com/othyn/go-calendar/assets/7256684/3456c2f8-3ff8-4107-9df5-6353bea5a3a0">

The events correctly display, and even when changing local timezone, the events remain at the same time respective to the _local current_ timezone and _not_ translated from UTC (they remain at the same time regardless of timezone).

Whereas in Google Calendar, the timezone is set to UTC by Google and you cannot change it, going against RFC 5545:

<img width="645" alt="Screenshot 2024-02-04 at 23 55 24" src="https://github.com/othyn/go-calendar/assets/7256684/adeaea64-4d01-40e8-a4e1-f65eba8ac577">

Meaning if your timezone deviates from UTC, then Google will try to be 'helpful' and translate it into your local timezone, going against RFC 5545's defined handling of floating time.

The only solution to this would be something as built in #4 and #5, which is a complete re-write of the project to dynamically generate ICS files in _each and every_ timezone. Urgh.

**Comment by othyn** [comment]
There's a few comments on HackerNews that point to this being the case since at least 2019:

---

> nfriedly on March 27, 2019
> 
> Google Calendar does let you set the time zone in the 'More options' view, FWIW. I use it occasionally.
> 
> 	
> maxxxxx on March 27, 2019
> 
> I know. But I want no time zone. When I plan my Germany trip next week I want to see the 10 am meeting there at 10 regardless of where I am.
> It should operate like a good old paper calendar.
>
>
> rlpb on March 27, 2019
>
> The iCalendar standard does actually support that, but perhaps Google Calendar's UI does not. I suspect it does work internally, because it interoperates with iCalendar fairly well.
>
> The standard calls it a "floating" DATE-TIME:
>
> "The recipient of an iCalendar object with a property value consisting of a local time, without any relative time zone information, SHOULD interpret the value as being fixed to whatever time zone the "ATTENDEE" is in at any given moment."

https://news.ycombinator.com/item?id=19501955

---

> funcDropShadow on March 27, 2019
> 
> For that you would like to have an explicit "local time zone" option. I had recently similar in application I am > developing, with datetime of which we are not sure in which time zone they are. That information might arrive later.
> 
> 	
> maxxxxx on March 27, 2019
> 
> On my Palm Pilot there was a “floating” time zone that behaved like this. But I can’t find this feature in any modernapp. Outlook doesn’t have it either.
> 
> 	
> kalleboo on March 27, 2019
> 
> Apple's calendar has it on macOS, but not on iOS... (it's annoying how mobile everything always has to be so neutered)
> 
> 	
> Nemo157 on March 27, 2019
> 
> And yet iOS calendar is also more powerful in that you can set different time zones on the start and end of a single > event, which macOS calendar will show but not let you edit.

https://news.ycombinator.com/item?id=19501832

---

I can attest to that last point. There are so many features that iOS Calendar can do that macOS Calendar cannot. Such as changing the starting location when adjusting the travel time... always have to swap to my iPhone to update calendar entries because the macOS version doesn't have feature parity.

**Comment by othyn** [comment]
There are also _many_ support threads such as [this one](https://support.google.com/calendar/thread/3400482/is-there-a-way-to-set-remove-time-zone-as-default-in-google-calendar?hl=en) on the Google support forums in which have been asking for this 'feature' for years.

**Comment by Boby360** [comment]
I assume this is not up to date with your most recent commits, but just wanted to show that my S23 Samsung calendar shows/previously showed my timezone and the GMT time window. 
![Screenshot_20240204_170124_Calendar](https://github.com/othyn/go-calendar/assets/1607258/5c62ff96-8cf6-4497-9023-846b58c66f02)

**Comment by Boby360** [comment]
> I've just done another test. No matter what, Google Calendar will _always_ set the timezone of the calendar subscription to `(GMT+00:00) Coordinated Universal Time` with the field to change it being non-editable. Whereas, Apple Calendar correctly adheres to RFC 5545 and _doesn't_ impose a timezone on the subscribed calendar.
> 
> <img alt="Screenshot 2024-02-04 at 23 48 35" width="1389" src="https://private-user-images.githubusercontent.com/7256684/302142961-3456c2f8-3ff8-4107-9df5-6353bea5a3a0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDcxNzY3MzEsIm5iZiI6MTcwNzE3NjQzMSwicGF0aCI6Ii83MjU2Njg0LzMwMjE0Mjk2MS0zNDU2YzJmOC0zZmY4LTQxMDctOWRmNS02MzUzYmVhNWEzYTAucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDIwNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAyMDVUMjM0MDMxWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MWZiZjJlMWM0ODgwZTM0MDRmMjM3ZDMyMDdjMjFkNmMzNzI0ZmQyYmNkNTA2ODYwOTM3NGM0Njc2YTI0NjQzMSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.0X9X4kL7l0ZehRN9c9jrnwmYg6I-nmXJYJZyTM5HIGQ">
> The events correctly display, and even when changing local timezone, the events remain at the same time respective to the _local current_ timezone and _not_ translated from UTC (they remain at the same time regardless of timezone).
> 
> Whereas in Google Calendar, the timezone is set to UTC by Google and you cannot change it, going against RFC 5545:
> 
> <img alt="Screenshot 2024-02-04 at 23 55 24" width="645" src="https://private-user-images.githubusercontent.com/7256684/302143081-adeaea64-4d01-40e8-a4e1-f65eba8ac577.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDcxNzY3MzEsIm5iZiI6MTcwNzE3NjQzMSwicGF0aCI6Ii83MjU2Njg0LzMwMjE0MzA4MS1hZGVhZWE2NC00ZDAxLTQwZTgtYTRlMS1mNjVlYmE4YWM1NzcucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDIwNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAyMDVUMjM0MDMxWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YWUxZWQwMmQxOTMwYzA4NjIzNWZjZjAyNDJjNTA4NmEzMzMzNjQ2OGQ2N2RiNTk5MmQzNzBjMTg4N2ZlMzMyMiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.tjUothsQVY7WWfmTFgDsuDD7pI4UDxRJ2c5wcHxhX_Q">
> Meaning if your timezone deviates from UTC, then Google will try to be 'helpful' and translate it into your local timezone, going against RFC 5545's defined handling of floating time.
> 
> The only solution to this would be something as built in #4 and #5, which is a complete re-write of the project to dynamically generate ICS files in _each and every_ timezone. Urgh.

Instead of generating multiple ICS files, the ICS file should be able to have time per each timezone within the ICS.
This still seems REAL stupid, although I would think a handful of timezones would probably cover the 99%?

Something like, -10 (Hawaii) then -8 to -5(Most of North/South America and then -1 to +2(EU and whatnot) then some japan, India, Australia and Newzealand?

I recall reading an article about how timezones (without all of this calendar stuff) is crazy complicated to have every timezone perfect.

**Comment by Boby360** [comment]
This is supposed to auto-update right?

I manually deleted and added it, and it shows the correct time and the abbreviated title!!!

**Comment by othyn** [comment]
> Instead of generating multiple ICS files, the ICS file should be able to have time per each timezone within the ICS.
This still seems REAL stupid, although I would think a handful of timezones would probably cover the 99%?

I didn't realise this! That could be a nice interim fix, I may have a play with doing that. Seems like covering the 8 below may gather the most coverage?

1. `UTC (Coordinated Universal Time)`: This serves as the reference time zone and is commonly used in international contexts.
2. `America/New_York`: Represents Eastern Time in the United States.
3. `America/Los_Angeles`: Represents Pacific Time in the United States.
4. `Europe/London`: Represents Greenwich Mean Time (GMT) in the United Kingdom.
5. `Europe/Berlin`: Represents Central European Time (CET) in mainland Europe.
6. `Asia/Tokyo`: Represents Japan Standard Time (JST) in Japan.
7. `Asia/Shanghai`: Represents China Standard Time (CST) in China.
8. `Australia/Sydney`: Represents Australian Eastern Standard Time (AEST) in Australia.

> I manually deleted and added it, and it shows the correct time and the abbreviated title!!!

<img src="https://media1.tenor.com/m/hyVt5PDLlzcAAAAC/futurama-good-news-everyone.gif" alt="Futurama good news everyone" />

> I forgot to point to that being a localized issue and not an overall ICS issue... ooops.

Haha no worries. Odd that it would only happen in the Community Day ICS file... such a bizarre problem all round...

**Comment by othyn** [comment]
> This is purely based on whatever chatgpt mentioned a couple times. Didn't validate and would take with a grain of salt. (the multi timezone data per ICS file)

Ahhhh okay. Yeah having a look at the RFC and some SO posts on this, it doesn't appear to be actually possible in practice, its one `DTSTART` and `DTEND` per `VEVENT` block. Oh well, back to the drawing board!

**Comment by othyn** [comment]
> Has this ever been resolved? Ive tried to use it today via Google Calendar, and manually importing the iCal into a Discord bot. The time is always +1 extra. 
> 
> 
> 
> Raids are always 18:00 local time, but in Google its still saying 19:00, in the Discord Bot to organise events, also says 19:00

Unfortunately not... I've tried just about everything at this point and nothing will get Google calendar (and now apparently Discord too) to interpret the ICS file properly, specifically when subscribed to. And at this point, I'm convinced it's a GCal bug.

The ICS spec says if you don't define a time zone, it should be interpreted in the users local calendar time. Apple (iOS, macOS, etc.) Calendar does this correctly, along with Microsoft too (Outlook, etc.) They both correctly adhere to the spec, even for subscribed calendars.

However, Google doesn't. They do if you import the calendar statically/manually from a download of the ICS file, but if you subscribe to it, for some reason it behaves differently. It assumes a time zone and removes the option to specify a time zone override, like you can when manually imported. But ironically, you then dont need the override, as it's interpreted correctly.

The only option is a complete rewrite of the entire repo to fix GCal, something similar to the serverless PR, to allow you to select a time zone and what event types you want and store that as a sequence of query parameters for the calendar URL. Basically move from static generation and relying on the client behaviour to offload compute locally, to dynamic generation for each client specific calendar server side, with aggressive daily caching. This way instead of publishing static calendars with no TZ definition, which should work according to the spec, they'd always be rendered server side/dynamically for the TZ you select.

My only concern on going that route and utilising something like CF's free tier for it, is that the repo gets a huge amount of hits, in the hundreds of thousands to millions a month, as the calendar clients are instructed to check daily for updates, as aligns with the cadence of updates from LeekDuck. It's an unavoidable requirement. And as we've seen more recently with CF on hacker news, I'm not sure they'd want it on the free tier for long. At the moment, it's all statically generated so that I can utilise GH pages/releases to host the static assets for free. I'm not even sure a cheap £5/mo DO droplet would do it, although I don't mind taking on that sort of cost or equivalent. I just love the simplicity of the deployment/hosting pipeline at the moment haha

**Comment by James-LSK** [comment]
Spitballing, may have missed it if it was mentioned in the thread, but a workaround could be just setting a timezone per calendar and having a separate link to each time zones for users to subscribe to. Of course it's not perfect i.e. for travel, but I think it would be sufficient for most users.

**Comment by othyn** [comment]
Its a good suggestion, and thanks for sharing, but yeah we've already explored it:

https://github.com/othyn/go-calendar/issues/4#issuecomment-1374480627

In summary, it generates far too many files at the moment for each timezone (around 4,000), and although they are small (on a few KB), I don't want to annoy GitHub by publishing a crazy amount of assets given the volume of hits each day. I'm currently utilising the static site generation and hosting to do this all for free. Don't bite the hand that feeds you and all that...

There was an option to use the free CF service workers tier instead, as someone made a brilliant PR for it to generate the calendars dynamically on the fly for a given timezone. However, there have since been many horror stories of CF closing out free tiers for 'high use' accounts and demanding payment (£££££), with terrible enterprise support. I'm usually a strong advocate of CF, but that really put me off.

However, I'm perhaps okay to revisit it for a select number of 'core' timezones (which was someones suggestion before), maybe 10 or 20 which cover one of each +/- 1 hour, which I guess will be a max of 48 calendar sets (to cover +/- 24 hours), so around ~25 (calendars) x 48 (timezones)... 1,200 calendar sets. Okay maybe not... maybe it really will have to be a small subset of timezones, just due to the volume of files, like maybe 10, as even 10 timezones is ~250 calendars.

I do also need to add a note to the website around the problems with Google Calendar and not abiding by RFC calendar subscription behaviour, so its more visible.

**Comment by eethomp** [comment]
Hi, this project is amazing in an overloaded month like this one! I encountered the Google bug, wandered over here, and read over the thread. Sadly my phone is Android so while it behaves on my MacBook, I don't use Apple Calendar. 

I downloaded the Community Day iCal and noticed that the DTSTAMP fields are in ISO 8601 format with the Z at the end to specify UTC with no offset, but the DTSTART and DTEND don't have any UTC offset information. 

I tried editing the commumity Day iCal file to make the times consistent and importing to Google. If I take the Z off of DTSTAMP so nothing has a UTC offset, it imports in my local time zone. Apple Calendar also reads the file with no offsets in local time and the events are 2-5PM. If I put a Z onto everything it's still bugged.

It appears that at least when an iCal file is directly imported, Google is setting the DTSTART and DTEND relative to the DTSTAMP.  I don't know if that's how iCal is supposed to work or whether it acts the same for the feed, but perhaps it's a helpful observation.
